app/Services/AWS.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Rogue\Services;
+
+use finfo;
+use Illuminate\Contracts\Filesystem\Filesystem;
+use Illuminate\Filesystem\FilesystemManager;
+use Symfony\Component\HttpFoundation\File\MimeType\ExtensionGuesser;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+
+class AWS
+{
+    /**
+     * The Amazon S3 file system.
+     * @see https://laravel.com/docs/5.2/filesystem
+     * @var Filesystem
+     */
+    protected $filesystem;
+
+    public function __construct(FilesystemManager $filesystem)
+    {
+        $this->filesystem = $filesystem->disk('s3');
+    }
+
+    /**
+     * Store a reportback item (image) in S3.
+     *
+     * @param \Symfony\Component\HttpFoundation\File\UploadedFile|string $file
+     *  File object, or a base-64 encoded data URI
+     * @param string $filename - Filename to write image to
+     *
+     * @return string - URL of stored image
+     */
+    public function storeReportbackItem($file, $filename)
```

Comment:
It might make sense to just call this method `storeImage` since it's not really specific to reportback items.